xdsl/transforms/riscv_register_allocation.py

```diff
@@ -89,6 +133,149 @@ def allocate_registers(self, module: ModuleOp) -> None:
                     self.idx += 1
 
 
+class RegisterLiveInterval(LiveRange):
+    """
+    Represents an enhanced live range with additional information for register allocation.
```

Comment:
Would be nice to expand this a lot, or add more inline comments. What's an abstract stack location? what's used?